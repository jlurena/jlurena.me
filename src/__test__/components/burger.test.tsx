import React from 'react';
import renderer from 'react-test-renderer';
import Burger from '../../js/components/burger';

describe('<Burger/>', () => {
  const onClickSpy = jest.fn();
  const component = renderer.create(
    <Burger onClick={onClickSpy} />
  );

  const tree = component.toJSON();

  test('Rendering', () => {
    expect(tree).toMatchSnapshot();
  });

  test('onClick handler gets called', () => {
    component.root.findByType('button').props.onClick();
    expect(onClickSpy).toHaveBeenCalledTimes(1);
  });
});
