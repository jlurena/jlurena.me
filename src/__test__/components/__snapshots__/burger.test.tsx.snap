// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Burger/> Rendering 1`] = `
<div
  className="burgerContainer"
>
  <button
    aria-label="Menu"
    className="burger"
    onClick={[MockFunction]}
    type="button"
  >
    <div />
    <div />
    <div />
    <div />
  </button>
</div>
`;
